pkg/ring/lifecycler.go

```diff
@@ -778,6 +781,16 @@ func (i *Lifecycler) SetFlushOnShutdown(flushOnShutdown bool) {
 	i.flushOnShutdown.Store(flushOnShutdown)
 }
 
+// SkipUnregister returns if unregistering is skipped on shutdown.
+func (i *Lifecycler) SkipUnregister() bool {
+	return i.skipUnregister.Load()
+}
+
+// SetSkipUnregister enables/disables unregistering on shutdown.
```

Comment:
This comment is slightly misleading, because by passing "true", we're actually disabling the unregister. (Why I suggested the rename)